### PR TITLE
vimPlugins: prefer self over super and, add dependencies for completion-*, gitsigns-nvim, and telescope-* plugins

### PR DESCRIPTION
**Pull request by malob**
###### Motivation for this change

It's nice when (Neo)Vim plugins installed via Nix include required plugin dependencies.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 119977` at 25f635e1 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built successfully:</summary>
  <ul>
    <li>spacevim</li>
    <li>vimPlugins.telescope-fzy-native-nvim</li>
  </ul>
</details>

File: pkgs/misc/vim-plugins/overrides.nix [hunk 1]
```diff
@@ -442,6 +459,18 @@ self: super: {
...
+  telescope-nvim = super.telescope-nvim.overrideAttrs (old: {
+    dependencies = [ self.plenary-nvim self.popup-nvim ];
+  });
+
+  telescope-symbols-nvim = super.telescope-symbols-nvim.overrideAttrs (old: {
+    dependencies = [ self.telescope-nvim ];
+  });
+
+  telescope-z-nvim = super.telescope-z-nvim.overrideAttrs (old: {
+    dependencies = [ self.telescope-nvim ];
```
Comment (SuperSandro2000) [inline pkgs/misc/vim-plugins/overrides.nix:471]:
```suggestion
    dependencies = with super; [ plenary-nvim popup-nvim ];
  });

  telescope-symbols-nvim = super.telescope-symbols-nvim.overrideAttrs (old: {
    dependencies = with super; [ telescope-nvim ];
  });

  telescope-z-nvim = super.telescope-z-nvim.overrideAttrs (old: {
    dependencies = with super; [ telescope-nvim ];
```

File: pkgs/misc/vim-plugins/overrides.nix [hunk 2]
```diff
@@ -415,10 +427,15 @@ self: super: {
...
-    dependencies = [ self.sql-nvim ];
+    dependencies = [ self.sql-nvim self.telescope-nvim ];
+  });
+
+  telescope-fzf-writer-nvim = super.telescope-fzf-writer-nvim.overrideAttrs (old: {
+    dependencies = [ self.telescope-nvim ];
   });
 
   telescope-fzy-native-nvim = super.telescope-fzy-native-nvim.overrideAttrs (old: {
+    dependencies = [ self.telescope-nvim ];
```
Comment (SuperSandro2000) [inline pkgs/misc/vim-plugins/overrides.nix:438]:
```suggestion
    dependencies = with super; [ telescope-nvim ];
```

File: pkgs/misc/vim-plugins/overrides.nix [hunk 3]
```diff
@@ -415,10 +427,15 @@ self: super: {
     });
 
   telescope-frecency-nvim = super.telescope-frecency-nvim.overrideAttrs (old: {
-    dependencies = [ self.sql-nvim ];
+    dependencies = [ self.sql-nvim self.telescope-nvim ];
+  });
+
+  telescope-fzf-writer-nvim = super.telescope-fzf-writer-nvim.overrideAttrs (old: {
+    dependencies = [ self.telescope-nvim ];
```
Comment (SuperSandro2000) [inline pkgs/misc/vim-plugins/overrides.nix:434]:
```suggestion
    dependencies = with super; [ telescope-nvim ];
```

File: pkgs/misc/vim-plugins/overrides.nix [hunk 4]
```diff
@@ -415,10 +427,15 @@ self: super: {
     });
 
   telescope-frecency-nvim = super.telescope-frecency-nvim.overrideAttrs (old: {
-    dependencies = [ self.sql-nvim ];
+    dependencies = [ self.sql-nvim self.telescope-nvim ];
```
Comment (SuperSandro2000) [inline pkgs/misc/vim-plugins/overrides.nix:430]:
```suggestion
    dependencies = with super; [ sql-nvim telescope-nvim ];
```

File: pkgs/misc/vim-plugins/overrides.nix [hunk 5]
```diff
@@ -258,6 +266,10 @@ self: super: {
     configurePhase = "cd plugins/nvim";
   });
 
+  gitsigns-nvim = super.gitsigns-nvim.overrideAttrs (old: {
+    dependencies = [ self.plenary-nvim ];
```
Comment (SuperSandro2000) [inline pkgs/misc/vim-plugins/overrides.nix:270]:
```suggestion
    dependencies = with super; [ plenary-nvim ];
```

File: pkgs/misc/vim-plugins/overrides.nix [hunk 6]
```diff
@@ -125,15 +125,23 @@ self: super: {
     '';
   });
 
+  completion-buffers = super.completion-buffers.overrideAttrs (old: {
+    dependencies = [ self.completion-nvim ];
+  });
+
   completion-tabnine = super.completion-tabnine.overrideAttrs (old: {
+    dependencies = [ self.completion-nvim ];
```
Comment (SuperSandro2000) [inline pkgs/misc/vim-plugins/overrides.nix:133]:
```suggestion
    dependencies = with super; [ completion-nvim ];
```

File: pkgs/misc/vim-plugins/overrides.nix [hunk 7]
```diff
@@ -125,15 +125,23 @@ self: super: {
     '';
   });
 
+  completion-buffers = super.completion-buffers.overrideAttrs (old: {
+    dependencies = [ self.completion-nvim ];
```
Comment (SuperSandro2000) [inline pkgs/misc/vim-plugins/overrides.nix:129]:
```suggestion
    dependencies = with super; [ completion-nvim ];
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by malob** [comment]
Hey @SuperSandro2000, thanks so much as always for the quick review.

Did you ever happen to get a chance to read my comment here?: https://github.com/NixOS/nixpkgs/pull/119356#issuecomment-819019006

I suspect I'm just missing something, but if I'm not, wouldn't it be better policy to default to using `self` when specifying these dependencies rather than `super`?

**Comment by SuperSandro2000** [comment]
> Did you ever happen to get a chance to read my comment here?: [#119356 (comment)](https://github.com/NixOS/nixpkgs/pull/119356#issuecomment-819019006)

I think I have read it and forgot about it again. I just looked at the file and since most are using super I thought that is the correct way. I think you should update all super to self to prevent that in the future.

> I suspect I'm just missing something, but if I'm not, wouldn't it be better policy to default to using self when specifying these dependencies rather than super?

Yeah, it would.

**Comment by malob** [comment]
Cool. I'll add a commit that makes that change.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 119977` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
18 packages built:
</summary>
<ul>
<li>spacevim</li>
<li>vimPlugins.completion-buffers</li>
<li>vimPlugins.completion-tabnine</li>
<li>vimPlugins.completion-treesitter</li>
<li>vimPlugins.ensime-vim</li>
<li>vimPlugins.gitsigns-nvim</li>
<li>vimPlugins.ncm2-jedi</li>
<li>vimPlugins.skim-vim</li>
<li>vimPlugins.telescope-frecency-nvim</li>
<li>vimPlugins.telescope-fzf-writer-nvim</li>
<li>vimPlugins.telescope-fzy-native-nvim</li>
<li>vimPlugins.telescope-nvim</li>
<li>vimPlugins.telescope-symbols-nvim</li>
<li>vimPlugins.telescope-z-nvim</li>
<li>vimPlugins.vim-addon-nix</li>
<li>vimPlugins.vim-addon-sql</li>
<li>vimPlugins.vim-addon-xdebug</li>
<li>vimPlugins.vimshell-vim</li>
</ul>
</details>

**Pull request merged**